STIPS/STIP-009.md

```diff
@@ -0,0 +1,240 @@
+- Allow `operator`(s) to initialize the StreamingFeeModule
+- Allow `owner` and `methodologist` to split streaming fees
+- Allow `owner` to update the streaming fee
+- Allow `owner` to update the streaming fee split
+- Allow `owner` to update the streaming fee recipient
+
+**TradeExtension**
+
+- Allow `owner` to enable functionality of TradeModule with only a state change and no contract deployment
+- Allow `operator`(s) to initialize the TradeModule
+- Allow privileged `operator`(s) to perform trades on a DEX
+- Allow `owner` to restrict assets the privileged `operator`(s) can trade into with an asset whitelist
+
+## User Flows
+
+### ManagerFactory.create()
+
+![ManagerFactory create](../assets/stip-009/image2.png "")
+
+An `owner` wants to create a new Set Token with a BaseManagerV3 smart contract manager.
+
+1. The `owner` calls create() passing in parameters to create a Set Token, parameters for the permissioning on BaseManagerV3, and the desired extensions. Specifically,
+
+    - components: List of addresses of components for initial positions
+    - units: List of units for initial positions
+    - name: Name of SetToken
+    - symbol: Symbol of SetToken
+    - owner: The address of the `owner`
+    - methodologist: The address of the `methodologist`
+    - operators: List of addresses of the `operator`(s)
+    - assets: List of addresses of assets for initial asset whitelist
+    - extensions: List of addresses of global extensions to be enabled
+
+2. A Set Token is deployed using SetTokenCreator
+3. A BaseManagerV3 is deployed with the ManagerFactory as the temporary `owner` until after initialization
+4. Initialization parameters for the Set Token are stored on the Factory
+5. The Set Token and Manager are put in pending state on the Factory
+
+### ManagerFactory.migrate()
+
+![ManagerFactory migrate](../assets/stip-009/image3.png "")
+
+An `owner` wants to migrate an existing Set Token to a BaseManagerV3 smart contract manager.
+
+1. The `owner` calls migrate() passing in the Set Token address, parameters for the permissioning on BaseManagerV3, and the desired extensions. Specifically,
+
+    - owner: The address of the `owner`
+    - methodologist: The address of the `methodologist`
+    - operators: List of addresses of the `operator`(s)
+    - assets: List of addresses of assets for initial asset whitelist
+    - extensions: List of addresses of global extensions to be enabled
+
+2. A BaseManagerV3 is deployed with the ManagerFactory as the temporary `owner` until after initialization
+3. Initialization parameters for the Set Token are stored on the Factory
+4. The Set Token and Manager are put in pending state on the Factory
+
+### ManagerFactory.initialize()
+
+![ManagerFactory initialize](../assets/stip-009/image4.png "")
+
+An `owner` wants to enable all extensions and initialize all corresponding modules.
+
+1. The `owner` calls initialize() passing in the parameters for initializing modules and extensions
```

Comment:
Remember this doesn't have to be the owner (esp since the owner could be a multi-sig), we want this to just be whatever the deployer address is